### ISSUE TITLE
AP not improving when finetuning on VAT

### ISSUE THREAD

**Issue by zdai257** [fkryan/gazelle#16]
Hi,

Thanks for sharing the source code. It's a truly inspiring work.

When I tried to reproduce the training, it converges well on GazeFollow. AUC and l2 seem good when finetuning VAT, but the AP stagnated at 0.64 throughout all epochs with either sample rate = 6 or 1. I tried to increase the 'inout lr' from 1e-2 to 4e-2, but no effects on AP. Any suggestions?

**Comment by fkryan** [comment]
Hi @zdai257,

In my local environment (RTX4090, Pytorch 2.6.0+cu118), I don't encounter this issue for VAT training for the ViT-L or ViT-B model. However, as I was refactoring my code for the release I did notice some sensitivity to the initialization of the added in/out parameters. Since your AP stagnates at 0.64, my guess is the model is stuck predicting 1 for the in/out score for every frame. I would try lowering the in/out learning rate (e.g. to 1e-3 or 1e-4) or changing the initialization (e.g. try a different random seed, change the order or method of the initialization of the in/out parameters in model.py). Please let me know if any of that works!

Also, the VAT training should be initialized from GazeFollow pretrained checkpoint using the '--init_checkpoint' argument, so be sure that is happening correctly.

**Comment by zdai257** [comment]
Changing the inout lr to either 1e-3 or 1e-4 solves the problem. Thank you!
